### ISSUE TITLE
ngSubmit when built with --prod flag not preventing form default behavior

### ISSUE THREAD

**Issue by LegendaryNIK** [angular/angular#21173]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
</code></pre>

## Current behavior
I've bumped in some strange error working on my project. 
When I build it with --prod flag, (submit) and (ngSubmit) in my forms just stop preventing default form behavior and page reloads after submition. 
Also, passing event and preventing it in onSubmit function won't give any result. 
FormsModule and ReactiveFormsModule are imported.

When I build without --prod flag, everything works well

Any workarounds for this? 

## Expected behavior
Submitting form won't reload page.

## Minimal reproduction of the problem with instructions
1. Download Angular reactive form demo https://angular.io/generated/zips/reactive-forms/reactive-forms.zip
2. Run npm install / ng build --prod
3. Open built app
4. Try to submit form

## Environment

<pre><code>
Angular version: 4.3.1

Browser:
Tested in Chrome and Firefox on Windows and Linux

**Comment by justerror** [comment]
same issue on various builds on Angular 5+

**Comment by GarryBrown** [comment]
Same issue (+validation form doesn't work) with --prod flag. In dev - it works right. 
Version 5.1.2

**Comment by thekiba** [comment]
We have the same problem, and it appeared between 22 December 02:00 and 23 December 22:00 (UTC). Angular version 5.0.1, Angular cli version 1.5.0.

**Comment by julianobrasil** [comment]
Something is going on in some cases and I'm not really sure why I haven't noticed it before in any of my projects in production. In fact, they seem to be working just fine, but when I build a test project this issue comes up.

There are issues also in Material2 repo (https://github.com/angular/material2/issues/9120) and in angular-cli repo (https://github.com/angular/angular-cli/issues/8997) likely related to this.

> (...) I could reproduce it in a test app, available in heroku. In the images below you can notice that not just the label stops working, **but also the interpolation** (see the gifs, just beneath the inputs). It doesn't seems to be related just to material... maybe `@angular/cli`'s `build-optimizer` flag has something to do with it. I noticed the problem using Chrome and Firefox (Windows 10)

### without --prod

![image](https://media.giphy.com/media/l0HUqJQsXldkSGwpy/giphy.gif)

### with --prod
![image](https://media.giphy.com/media/l49JPlaGVUnsSvj0c/giphy.gif)

**Comment by julianobrasil** [comment]
This @cuiliang's suggestion (https://github.com/angular/angular-cli/issues/8997#issuecomment-353957698) worked **partially** for me (in my tiny test project)

```
ng build --aot --build-optimizer true --output-hashing all --extract-css true --name
         d-chunks false --sourcemaps false --environment prod
```

[edited]: but the chunks size increased a lot 😸 - cannot use it in production

**Comment by GarryBrown** [comment]
@julianobrasil  we have done that :D

**Comment by thekiba** [comment]
```
// file: /node_modules/@angular/cli/models/webpack-configs/production.js

- mangle: true
+ mangle: false
```

This solves the problem and made it possible to build the project.

**Comment by julianobrasil** [comment]
@GarryBrown, as suggested by @filipesilva in https://github.com/angular/angular-cli/issues/8997, the problem seems to be related to `uglify-es@3.3.2`:

So, temporarily:

1 - add `uglify-es@3.2.2` to dependencies array in package.json
2 - delete package-lock.json, node_modules and dist
3 - npm install

Or, to linux users:
```
npm i uglify-es@3.2.2 --save-exact && rm -rf package-lock.json node_modules && npm i && npm ls uglify-es
```

**Comment by maxkorz** [comment]
https://github.com/mishoo/UglifyJS2/issues/2664

**Comment by Alekcei** [comment]
Sorry, I was wrong.
```
app5test@0.0.0 /home/alekcei/workspase/BID
├─┬ @angular/cli@1.6.2
│ └─┬ uglifyjs-webpack-plugin@1.1.4
│   └── uglify-es@3.2.2  deduped
└── uglify-es@3.2.2 

```
Yes, it's working now

**Comment by muchrm** [comment]
thank @julianobrasil 

> ng build --aot --build-optimizer true --output-hashing all --extract-css true --name
         d-chunks false --sourcemaps false --environment prod
it's work for me

**Comment by julianobrasil** [comment]
@muchrm , the chunks sizes are smaller if you follow https://github.com/angular/angular/issues/21173#issuecomment-354047384

**Comment by thiagovito** [comment]
@julianobrasil I was only able to fix the problem by making the change that @thekiba mentioned on his [comment](https://github.com/angular/angular/issues/21173#issuecomment-353969005)

**Comment by julianobrasil** [comment]
It looks like the cli team is about to release a patchwork.

**Comment by IgorMinar** [comment]
@filipesilva could this be due to build-optimizer?

@LegendaryNIK @julianobrasil can you try building the project with `--prod --no-build-optimizer` and report here?

**Comment by orizens** [comment]
@IgorMinar @filipesilva @LegendaryNIK @julianobrasil 
```ng serve --prod --no-build-optimizer``` works as well.  
I had a similar error with the latest build of http://github.com/orizens/echoes-player

**Comment by julianobrasil** [comment]
@IgorMinar, I think the problem was identified by @filipesilva in https://github.com/angular/angular-cli/issues/8997#issuecomment-354125877

Anyway, I pinned the  version to the problematic one and I can confirm that the result with and without `--no-build-optimizer` is the bugged production bundles.

**Comment by IgorMinar** [comment]
we just released @angular/cli@1.6.3 which works around a problem in uglify that causes this. see angular/angular-cli#8997
